proxy/envoy/agent.go

```diff
@@ -44,39 +46,21 @@ type instance struct {
 }
 
 const (
-	// EnvoyConfigPath is the root directory for Envoy configuration
-	EnvoyConfigPath = "/etc/envoy/"
-
 	// EnvoyFileTemplate is a template for root config JSON
-	EnvoyFileTemplate = "envoy-rev%d.json"
+	EnvoyFileTemplate = "/envoy-rev%d.json"
 )
 
-// NewAgent creates a new instance.
-func NewAgent(binary string, mixer string) (Agent, error) {
-	// TODO mixer should be configured directly in the envoy filter
-	f, err := os.Create(EnvoyConfigPath + ServerConfig)
-	if err != nil {
-		return nil, err
-	}
-	_, _ = f.WriteString(fmt.Sprintf(`
-cloud_tracing_config {
-  force_disable: true
-}
-mixer_options {
-  mixer_server: "%s%s"
-}
-	`, OutboundClusterPrefix, mixer))
-
-	_ = f.Close()
-
+// NewAgent creates a new proxy instance agent for a config root
+func NewAgent(binary string, configRoot string) Agent {
 	return &agent{
-		binary: binary,
-		cmdMap: make(map[*exec.Cmd]instance),
-	}, nil
+		binary:     binary,
+		configRoot: configRoot,
+		cmdMap:     make(map[*exec.Cmd]instance),
+	}
 }
 
-func configFile(epoch int) string {
-	return fmt.Sprintf(EnvoyConfigPath+EnvoyFileTemplate, epoch)
+func configFile(config string, epoch int) string {
+	return fmt.Sprintf(config+EnvoyFileTemplate, epoch)
```

Comment:
any reason to not just have the EnvoyFileTemplate include a %s at the beginning and pass config as an arg here?